src/components/ReportStatus.jsx

```diff
@@ -0,0 +1,167 @@
+import React, { useState, useEffect } from "react";
+import { Formik, Form } from "formik";
+import { Link } from 'react-router-dom';
+import { GetResponseErrors } from "../utilities/CitysourcedResponseHelpers";
+import { GetReportComments } from '../services/authService';
+import FormContainer from './FormContainer';
+import Note from './Note';
+import Moment from 'react-moment';
+import SeButton from "./SeButton";
+import { Go, Routes } from "../Routing";
+
+
+const ReportStatus = (props, routeProps) => {
+	const { trackingNumber } = props.values;
+	const response = props.history.location.state;
+	const [comments, setComment] = useState([]);
+	const [commentLength, setcommentLength] = useState([]);
+
+	let alertMessage = '';
+	let errorStatusCode ='';
+	let reportId, reportDateCreated, reportDateUpdated, reportRequestType, address, status, isOpen;
+
+	useEffect(() => {
+		setcommentLength(3);
+		buildComments();
+	}, [response]);
+
+	  
+	const buildComments = async() =>{
+		try{
+			if(trackingNumber){
+				const commentResponse = await GetReportComments(trackingNumber)
+
+				if(commentResponse.ErrorsCount > 0){
+					const errorsReturned = GetResponseErrors(commentResponse);
+					throw new Error(errorsReturned);
+				}
+				else{
+					setComment(commentResponse.data.Results);
+				}
+			}
+		}
+		catch(ex){
+
+		}
+	}
+
+	const displayReportStatus = () => {
+		if(!response)
+		{
+			Go(props, Routes.GetReport)
+		}
+		else if (response.data.ErrorsCount > 0) {
+			const errorsReturned = GetResponseErrors(response);
+			errorStatusCode = response.data.Errors[0].StatusCode;
+			alertMessage = <Note>{errorsReturned}</Note>		
+		}
+		else{
+			const {Id, DateCreated, DateUpdated, RequestType, FormattedAddress, StatusTypeReadable, StatusTypeIsClosed} = response.data.Results;
+			reportId = Id;
+			reportDateCreated = DateCreated;
+			reportDateUpdated = DateUpdated;
+			reportRequestType = RequestType;
+			address = FormattedAddress;
+			status = StatusTypeReadable;
+			isOpen = (StatusTypeIsClosed ? 'closed': 'open')
+		}
+	}
+
+	const showMoreComments = (clickEvent) =>{
+		if (clickEvent.currentTarget.innerText === 'Show more...')
+		{
+			setcommentLength(comments.length);
+			clickEvent.currentTarget.innerText = 'Show less...';
```

Comment:
in this case, you should use `useState` instead of modifying the dom element directly.

Reply:
converted to useState